controller/src/main/java/io/pravega/controller/store/stream/ZKGarbageCollector.java

```diff
@@ -146,11 +146,16 @@ int getLatestBatch() {
                 }, gcExecutor)
                 // fetch the version and update it.
                 .exceptionally(e -> {
-                    if (Exceptions.unwrap(e) instanceof StoreException.WriteConflictException) {
+                    Throwable unwrap = Exceptions.unwrap(e);
+                    if (unwrap instanceof StoreException.WriteConflictException) {
                         log.debug("Unable to acquire guard. Will try in next cycle.");
                     } else {
                         // if GC failed, it will be tried again in the next cycle. So log and ignore.
-                        log.error("Exception thrown during Garbage Collection iteration for {}. Log and ignore.", gcName, e);
+                        if (unwrap instanceof StoreException.StoreConnectionException) {
+                            log.error("Store Connection Exception thrown during Garbage Collection iteration for {}.", gcName);
```

Comment:
If both log messages are `error` level, then do we even need to separate them? To determine the severity level, we need to reason about whether it is actionable.

Reply:
I will make both the logs at `warning`. They may be actionable or ignorable. This piece of code calls GC periodically irrespective of success or failure of previous iteration.